src/shared/resourcefetcher/httpResourceFetcher.ts

```diff
@@ -175,3 +176,17 @@ export async function getPropertyFromJsonUrl(
         }
     }
 }
+
+/**
+ * Retrieves sha-256 hash of the contents at a remote resource
+ * @param url url of the resource
+ * @param fetcher optional HTTP resource fetcher to use
+ * @returns hashed contents from the url if available or undefined
+ */
+export async function getContentHashFromUrl(url: string, fetcher?: HttpResourceFetcher): Promise<string | undefined> {
```

Comment:
Many things can be done with a `resourceFetcher.get()` result. But we wouldn't want to add `httpResourceFetcher.ts` functions for every operation that can be done on text. 

Instead, this is a signal that `HttpResourceFetcher` is too cumbersome to set up and use. So let's fix that instead of adding a special-case function just to hash the contents of a downloaded file. I.e. we could add a function here like:

```
export async function getFromurl(url: string, fetcher): ... {
    const resourceFetcher = fetcher ?? new HttpResourceFetcher(url, { showUrl: true })
    const result = await resourceFetcher.get()
    return result ? result : undefined
}
```

Then the call site looks like:

```
getStringHash(getFromurl(...))
```

This is a general principle that can/should be applied to all future code.

Reply:
should `url` at least be `string | vscode.Uri` (or even just `vscode.Uri` ) ?

Reply:
What advantage would there to be passing in a `vscode.URI`? I think we're probably fine with just a url `string` unless we eventually expand the scope of the resource fetcher

Reply:
There might be no immediate advantage. But that's usually the case when we start with string for Uris--then the pattern tends to bleed into the codebase over time. Not a blocker though.